packages/native/types/base.d.ts

```diff
@@ -73,9 +69,11 @@ type ReactNativeTags =
   | 'ViewPagerAndroid'
   | 'WebView'
 
-export type ReactNativeElements = {
-  [Tag in ReactNativeTags]: PropsOf<ReactNative[Tag]>
-}
+export type ReactNativeComponents = Pick<ReactNative, ReactNativeComponentNames>
+
+export type ReactNativeComponentProps<
+  ComponentName extends ReactNativeComponentNames
+> = ComponentPropsWithoutRef<ReactNativeComponents[ComponentName]>
```

Comment:
I noticed i was needing to extract the props out quite frequently, so i built a handy extractor construct.